packages/flutter/lib/src/rendering/view.dart

```diff
@@ -90,8 +91,11 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
     assert(value != null);
     if (configuration == value)
       return;
+    final double oldDevicePixelRatio = _configuration.devicePixelRatio;
     _configuration = value;
-    replaceRootLayer(_updateMatricesAndCreateNewRootLayer());
+    if (oldDevicePixelRatio != value.devicePixelRatio) {
```

Comment:
There are a decent number of properties  on view configuration: https://github.com/flutter/engine/blob/d24339f20462eb10d96df45ecc4c2c42f6fadd20/lib/ui/platform_dispatcher.dart#L1173 .

Is devicePixelRatio the only one that requires replacing the root layer? If there will be potentially more root-replacements required in the future it would be advantageous to document this a bit more extensively.

Reply:
Currently this is the only case that needs replacement of the root layer. Maybe @goderbauer can double check this.
I agree with that document this a bit more.

Reply:
While the devicePixelRatio is the only thing that currently required replacing the root layer, that's really an implementation detail of `ViewConfiguration.toMatrix` and not really known at this level of abstraction right here - and if we change that in the future we may forget about this check here.

To make this a little saner, we could maybe add an API to ViewConfiguration that takes an old config to ask it whether the transformation matrix as changed and only replace the root layer if it did. That way, creating the matrix and checking whether it needs to be re-created is all encapsulated in the ViewConfiguration class.

Reply:
Done